pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -494,6 +504,52 @@ func (c *Cluster) saveMonConfig() error {
 	return nil
 }
 
+func (c *Cluster) saveCsiClusterConfig() error {
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      csiConfigName,
+			Namespace: c.Namespace,
+		},
+	}
+	k8sutil.SetOwnerRef(c.context.Clientset, c.Namespace, &configMap.ObjectMeta, &c.ownerRef)
+
+	type csiClusterConfig struct {
+		ClusterID string   `json:"clusterID"`
+		Monitors  []string `json:"monitors"`
+	}
+	cc := make([]csiClusterConfig, 1)
+	cc[0].ClusterID = c.clusterInfo.FSID
+	cc[0].Monitors = []string{}
+	for _, m := range c.clusterInfo.Monitors {
+		cc[0].Monitors = append(cc[0].Monitors, m.Endpoint)
+	}
+
+	ccJson, err := json.Marshal(cc)
+	if err != nil {
+		return fmt.Errorf("failed to marshal csi cluster config. %+v", err)
+	}
+
+	configMap.Data = map[string]string{
+		csiConfigKey: string(ccJson),
+	}
+
+	if _, err := c.context.Clientset.CoreV1().ConfigMaps(c.Namespace).Create(configMap); err != nil {
```

Comment:
(Ignore this comment if multiple clusters need to be stored in the same configmap)

Instead of creating a new configmap that the csi driver understands, what would it take to teach the csi driver to look for the `rook-ceph-mon-endpoints` configmap that already exists? If the format of those endpoints isn't friendly to the csi driver, we could also just add another entry to that configmap instead of creating a new one. For example, could another "csi" entry be added to the `data` map for this configmap?
```
$ kubectl -n rook-ceph get cm rook-ceph-mon-endpoints -o yaml
apiVersion: v1
data:
  data: a=10.98.10.44:6789
  mapping: '{"node":{"a":{"Name":"minikube","Hostname":"minikube","Address":"10.0.2.15"}},"port":{}}'
  maxMonId: "0"
kind: ConfigMap
...
```

Reply:
I added a new top level key to the existing config map. I hope this satisfies the request as well as the csi driver needs.